CalcualtorLogic.js

```diff
@@ -107,27 +107,20 @@ class Calculator {
         return div
     };
     onButtonNumberClick(event) {
-        if (this.lastButtonClicked === "operator") this.resultElement = "";
+        if (this.lastButtonClicked === "operator") this.currentResult = "";
         const currentResultAndclickedNumButton = this.currentResult + event.target.dataset.number;
         this.currentResult = currentResultAndclickedNumButton;
         this.updateResultElement();
-
         // console.log(this.last)
     }
     updateResultElement() {
         this.resultElement.textContent = this.currentResult;
     }
     onOperatorButtonClick(event) {
-        this.operator = event.target.dataset.operator;
-        this.currentResult = 0;
-        this.currentResult = this.currentResult + event.target.dataset.operator;
+        const clkedOperButn = event.target.dataset.operator;
+        this.operator = clkedOperButn;
         this.lastButtonClicked = "opertaor"
-        this.resultElement = this.currentResult;
-        console.log(this.resultElement)
-
-        // console.log(this.operator)
-        // this.last = "foo"
-
+        this.lastValue = this.resultElement;
```

Comment:
If I am not wrong, I think you're trying to copy the `currentResult` to the `lastValue` 
if this is what your intention then that's good but your implementation is wrong.

I said to you long time ago, this time we will not use the UI to get the current state of the calculator.
in other words, we will not get the current result from the result div `this.resultElement`.

I don't want you to use HTML elements to get the current result 
you need to search for something else available for you on the class to get the current result of the result div

Reply:
previously I tried so much by using :
 updateResultElement()
but when I was clicking on the operator buttons always I was getting an error messages so I did not use it 

I do not why there is no error messages in   onButtonNumberClick method 
thanks

Reply:
I don't understand why you so eager to use `updateResultElement` 
the `updateResultElement` is used to copy the value of `currentResult` to `resultElement` 
this method responsible for updating the user interface (UI) so I don't know how it's going to help you here?

I want to perform 1 + 2 
I click on 1 then + 2

when I click on + button, what should happen?????

at this point the UI is showing 1 

we need to copy this 1 which is showing in UI to another variable/property to allow entering the second operand 
we need to reset the UI 

we decided to clear the UI when a user enters the second operand 

so when he clicks on 1 + 
he continue to see 1 
as soon he clicks on 2, the UI is cleared and because the UI is cleared an empty string will be added to 2 and users will see 2 only not 12 


I said many times the UI need to be cleared, and we need to copy the UI value to another variable/property 
the element which represent the UI is saved as property on the class `resutlElement` 
Please don't every try to get a value from this property 
think of this object is locked somehow, and you can only set value to it, but you can't get values from it.

so Please, please don't go and try to get the current result from `resultElement` 


this what need to be done:
in the `onOperatorButtonClick` you need:
1. get the value of the clicked button and save it in a property 
2. copy the value of the current result to another variable (see the example above 1 + 2) 
3. update the value of the lastButtonClicked property to a value that you can assert on it later when you want to clear the UI if the last button was clicked is an operator (check if you have typos )

in the `onButtonNumberClick`:
1. check if the last button clicked is an operator and if it's then clear the UI

Reply:
I am not eager but unfortunately do not knowing what you are asking me to do or what the class calculator wants to work 
so I am writing any codes which maybe can work by chance
========
would you tell me in the previous implementation how we used the UI to get the current result that one which will not happen in this one 
===========
please why you are writing we want to copy and not writing we want to add although you are writing we want to perform 1+2 
=============
I do not understanding what is resetting the interface
what is coming in my mind the interface is that html and css codes which we used to build the calculator
so do we need to change some divs or some ids or class names or what 
===============
yes this happened if I was getting the value from resultElement 
==========
I have pushed some updates regarding for getting the current result from resultElement

what I have pushed is only about the first step
please let me know my mistakes before continuing the second and third steps 

thanks